pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -413,6 +416,10 @@ cdef class {{name}}HashTable(HashTable):
         cdef:
             khiter_t k
             {{c_type}} ckey
+
+        if self.uses_mask and checknull(key):
```

Comment:
does it matter what type of null we're looking at?

Reply:
Depends if we want to support multiple nulls in masked arrays. Currently it does not:

```
idx = Index([1, 2, pd.NA, pd.NA], dtype="Int64")
idx.get_loc(None)
idx.get_loc(np.nan)
```

Both match pd.NA

Reply:
Ah got you. This is interesting currently:

```
idx.get_loc(pd.NaT)
```

matches both NAs, but 

```
Index([1, 2, pd.NA, pd.NA, pd.NaT], dtype="Int64")
```
raises. Any suggestions here?

Reply:
is_valid_na_for_dtype exists for this purpose.  For other dtypes we do this in FooIndex.get_loc.  Might require some gymnastics in this case

Reply:
In general I would be ok with pushing responsibility to the caller. Have to do this for `` set_item`` and ``get_item`` anyway

Reply:
definitely reasonable, pls be sure to document this assumption

Reply:
Added a comment. I'll probably add some documentation in general when we are through with a MaskedEngine